File: test_data/code/python/1805-number-of-different-integers-in-a-string.py
class Solution:
    def numDifferentIntegers(self, word: str) -> int:
        res = set()
        cur_num = ''
        for char in word:
            if char.isdigit():
                cur_num += char
            else:
                if cur_num:
                    cur_num = cur_num.lstrip('0')
                    if cur_num:
                        res.add(cur_num)
                    cur_num = ''
        if cur_num:
            cur_num = cur_num.lstrip('0')
            if cur_num:
                res.add(cur_num)
        return len(res)

File: test_data/code/java/1805-number-of-different-integers-in-a-string.java
class Solution {
    public int numDifferentIntegers(String word) {
        String[] nums = word.split("[a-z]+");
        Set<String> set = new HashSet<>();
        for (String num : nums) {
            num = num.trim();
            if (!num.isEmpty()) {
                set.add(trimLeadingZeros(num));
            }
        }
        return set.size();
    }

    private String trimLeadingZeros(String num) {
        while (!num.isEmpty() && num.charAt(0) == '0') {
            num = num.substring(1);
        }
        return num;
    }
}